src/forest.cpp

```diff
@@ -24,17 +24,92 @@
 #include <iosfwd>            // for ostream
 #include <vector>            // for vector, allocator, operator==
 
+#include "libsemigroups/bipart.hpp"
 #include "libsemigroups/constants.hpp"  // for Max, operator!=, operator==
 #include "libsemigroups/exception.hpp"  // for LIBSEMIGROUPS_EXCEPTION
 #include "libsemigroups/types.hpp"      // for word_type
 
 #include "libsemigroups/detail/string.hpp"  // for operator<<
 
+// namespace forest {
+
+//! Returns a \ref normal_form_iterator pointing at the first normal
+//! form.
+//!
+//! Returns a const iterator pointing to the normal form of the first
+//! class of the congruence represented by an instance of ToddCoxeter.
+//! The order of the classes, and the normal form, that is returned are
+//! controlled by standardize(order).
+//!
+//! \returns A value of type \ref normal_form_iterator.
+//!
+//! \exceptions
+//! \no_libsemigroups_except
+//!
+// TODO(2): update the doc
+// TODO(2) This is probably not very efficient, better to have a custom range
+//
+// where the word_type is cached, and next just deletes the end of the word
+// and adds whatever's appropriate, rather than retracing the entire path
+// every time
+//
+// inline auto paths(Forest const& f) {
+//   rx::seq<Forest::node_type>() | rx::transform([&f](auto i) {
+//     word_type w;
+//     while (f.parent_no_checks(i) != UNDEFINED) {
+//       w.push_back(f.label_no_checks(i));
+//       i = f.parent_no_checks(i);
+//     }
+//     return w;
+//   });
+// }
+//}  // namespace forest
```

Comment:
Is this commented out deliberately?

Reply:
Yes, it seems not to be used anywhere and there's the TODO about improving it.